milli/src/search/query_tree.rs

```diff
@@ -645,7 +685,7 @@ fn create_matching_words(
                                 for synonym in synonyms {
                                     let synonym = synonym
                                         .into_iter()
-                                        .map(|syn| MatchingWord::new(syn, 0, false))
+                                        .flat_map(|syn| matching_word_cache.insert(syn, 0, false))
```

Comment:
same as before `flat_map` is obscure 😬

Reply:
Are you sure you don't want to use a `map` instead, and collect into an `Option<Vec<_>>` as you did before?
Honestly, I'd prefer to avoid the use of `flat_map`, and at least replace it with a `filter_map` or a `map_while`. 😬